perllib/FixMyStreet/Cobrand/Sutton.pm

```diff
@@ -804,4 +804,34 @@ sub waste_munge_small_items_data {
     $self->waste_munge_bulky_data($data);
 }
 
+=head2 waste_show_cancel_request
+
+Show the option to cancel a request for all users, if the report hasn't
+already been cancelled.
+
+=cut
+
+sub waste_show_cancel_request {
+    my ($self, $request_report) = @_;
+    return $request_report->state ne 'cancelled';
+}
+
+=head2 waste_can_cancel_request
+
+Allows cancelling a request if the user is staff or the person that
+made the request, and the report is not already cancelled.
+
+=cut
+
+sub waste_can_cancel_request {
```

Comment:
I'd probably have this as the default, rather than Sutton only, and add a feature flag check for something (`request_cancel_enabled` would match the bulky one), then it's easier to add anyone else in future.